modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/source/ExtensionMessageSource.java

```diff
@@ -646,6 +652,7 @@ protected ParameterValueResolver getParameterValueResolver() {
 
   @Override
   protected void doInitialise() throws InitialisationException {
+    validateConfigurationProviderIsNotExpression();
```

Comment:
Since I removed this validation from the base class (`ExtensionComponent`) I had to move it here.